File: December - 02/SakthiDharan01_total_of_distinct_values.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class SakthiDharan01_total_of_distinct_values {
    private static void solve() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String line = br.readLine();
        if (line == null || line.trim().isEmpty()) {
            return;
        }
        int n = Integer.parseInt(line.trim());
        int width = Integer.toBinaryString(n).length();
        StringBuilder sb = new StringBuilder();
        for (int i = 1; i <= n; i++) {
            String decStr = String.format("%" + width + "d", i);
            String octStr = String.format("%" + width + "o", i);
            String hexStr = String.format("%" + width + "X", i);
            String binStr = String.format("%" + width + "s", Integer.toBinaryString(i));
            sb.append(decStr).append(' ').append(octStr).append(' ').append(hexStr).append(' ').append(binStr).append('\n');
        }
        System.out.print(sb.toString());
    }

    public static void main(String[] args) throws Exception {
        solve();
    }
}

File: December - 02/SakthiDharan01_total_of_distinct_values.py
import sys

def solve():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    n = int(data[0])
    width = len(bin(n)) - 2
    for i in range(1, n + 1):
        dec_str = format(i, "d").rjust(width)
        oct_str = format(i, "o").rjust(width)
        hex_str = format(i, "X").rjust(width)
        bin_str = format(i, "b").rjust(width)
        print(dec_str, oct_str, hex_str, bin_str)

if __name__ == "__main__":
    solve()
